src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java

```diff
@@ -110,4 +135,77 @@ protected static List<Configuration> getModuleConfigsByIds(String... moduleIds)
         return getModuleConfigsByIds(CONFIGURATION, moduleIds);
     }
 
+    /**
+     * Performs verification of the file with the given file path against the whole google config.
+     *
+     * @param filePath file path to verify.
+     * @throws Exception if exception occurs during verification process.
+     */
+    public void verifyWithWholeGoogleConfig(String filePath) throws Exception {
+        final Configuration[] children = CONFIGURATION.getChildren();
+        final DefaultConfiguration root = new DefaultConfiguration(ROOT_MODULE_NAME);
+
+        for (Configuration child : children) {
+            root.addChild(child);
+        }
+
+        verifyWithItConfig(root, filePath);
```

Comment:
> verifyWithItConfig method in this abstract class, which will take in the entire config.

Why can't we replace `root` parameter with `CONFIGURATION`? The code above looks like it is not "Google" specific.

Reply:
we cannot directly pass the `CONFIGURATION` to the `verifyWithItConfig()` because `verifyWithItConfig()` uses [getActualViolationsForFile()](https://github.com/checkstyle/checkstyle/blob/aff00e1ab05cc59fc35158b99cf380ae1cabe3ca/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L481) under the hood and this method uses `createChecker()` inside it:

https://github.com/checkstyle/checkstyle/blob/aff00e1ab05cc59fc35158b99cf380ae1cabe3ca/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L484-L487

The `createChecker()` method requires the top level module name to be `root` otherwise it gives error. 

That's why we're getting the whole configuration, changing the top level module name and then passing it to the `verifyWithItConfig()`:

https://github.com/checkstyle/checkstyle/blob/aff00e1ab05cc59fc35158b99cf380ae1cabe3ca/src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java#L144-L153


This is also the same reason behind when using the `verifyWithGoogleConfigParser()` method, we create the config with the `root` as top level module:

https://github.com/checkstyle/checkstyle/blob/aff00e1ab05cc59fc35158b99cf380ae1cabe3ca/src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java#L163-L165

Reply:
> The createChecker() method requires the top level module name to be root otherwise it gives error.

This was resolved with the git diff I provided in discord.

Reply:
done added your changes

Reply:
> This is also the same reason behind when using the verifyWithGoogleConfigParser() method, we create the config with the root as top level module:

I have still not changed the code for `verifyWithGoogleConfigParser`, I could not find a way to create config which will have same Checker properties as our google config and I guess it is okay to keep this code as we're not checking against the whole config file, we're checking against only some specific modules

Reply:
As I mentioned in other comment, `verifyWithGoogleConfigParser` should be completely removed eventually, so it is ok that it doesn't do everything since I see it as the old/bad form of testing.

>  I could not find a way to create config which will have same Checker properties as our google config

Just to provide you clarity, what you are missing in the code is calling `CONFIGURATION.getProperties()`. You call get children which only returns the children modules. This is why the original code for `verifyWithWholeConfig` did not pull in the `Checker` properties either.

Reply:
I am resolving this. If there is any further questions or clarifications needed, please reach out in discord.

Reply:
> As I mentioned in other comment, verifyWithGoogleConfigParser should be completely removed eventually, so it is ok that it doesn't do everything since I see it as the old/bad form of testing.

I think we should keep this method for very large input files. For now lets keep this `until` message, we can discuss this later in discord

> Just to provide you clarity, what you are missing in the code is calling CONFIGURATION.getProperties(). You call get children which only returns the children modules. This is why the original code for verifyWithWholeConfig did not pull in the Checker properties either.

yep I found the way after I commented